ui/non_interactive.go

```diff
@@ -40,12 +40,24 @@ func (ui *NonInteractiveUI) PrintTable(table Table) {
 	ui.parent.PrintTable(table)
 }
 
-func (ui *NonInteractiveUI) AskForText(label string) (string, error) {
-	panic("Cannot ask for input in non-interactive UI")
+func (ui *NonInteractiveUI) AskForText(opts TextOpts) (string, error) {
+	if opts.ValidateFunc != nil {
+		isValid, err := opts.ValidateFunc(opts.Default)
+		if err != nil || !isValid {
+			return "", err
+		}
+	}
+	return opts.Default, nil
 }
 
-func (ui *NonInteractiveUI) AskForChoice(label string, options []string) (int, error) {
-	panic("Cannot ask for a choice in non-interactive UI")
+func (ui *NonInteractiveUI) AskForChoice(opts ChoiceOpts) (int, error) {
+	if opts.ValidateFunc != nil {
```

Comment:
Basic logic that default has to be between 0 and len(choices) is missing.

Reply:
Yeah, I was thinking if we are given the option to pass validateFunc then we should validate here or not. But yeah we can add the basic validations like default value should be in the range of `0 to len(opts.Choices) - 1`

Reply:
Added the validation.

Reply:
im not sure it makes sense to have ValidateFunc for choice, since by definition only one of the given choices can be chosen.